zcash_client_sqlite/src/lib.rs

```diff
@@ -1082,317 +1085,33 @@ extern crate assert_matches;
 
 #[cfg(test)]
 mod tests {
-    use prost::Message;
-    use rand_core::{OsRng, RngCore};
-    use rusqlite::params;
-    use std::collections::HashMap;
+    use zcash_client_backend::data_api::{WalletRead, WalletWrite};
 
-    #[cfg(feature = "unstable")]
-    use std::{fs::File, path::Path};
-
-    #[cfg(feature = "transparent-inputs")]
-    use zcash_primitives::{legacy, legacy::keys::IncomingViewingKey};
+    use crate::{
+        testing::{init_test_accounts_table_ufvk, network},
+        wallet::init::init_wallet_db,
+        AccountId, WalletDb,
+    };
 
-    use zcash_note_encryption::Domain;
+    #[cfg(feature = "unstable")]
     use zcash_primitives::{
         block::BlockHash,
-        consensus::{BlockHeight, Network, NetworkUpgrade, Parameters},
-        legacy::TransparentAddress,
-        memo::MemoBytes,
-        sapling::{
-            note_encryption::{sapling_note_encryption, SaplingDomain},
-            util::generate_random_rseed,
-            value::NoteValue,
-            Note, Nullifier, PaymentAddress,
-        },
+        consensus::{BlockHeight, Parameters},
         transaction::components::Amount,
-        zip32::{sapling::DiversifiableFullViewingKey, DiversifierIndex},
     };
 
-    use zcash_client_backend::{
-        data_api::{WalletRead, WalletWrite},
-        keys::{sapling, UnifiedFullViewingKey},
-        proto::compact_formats::{
-            self as compact, CompactBlock, CompactSaplingOutput, CompactSaplingSpend, CompactTx,
-        },
-    };
-
-    use crate::{
-        wallet::init::{init_accounts_table, init_wallet_db},
-        AccountId, WalletDb,
-    };
-
-    use super::BlockDb;
+    #[cfg(feature = "unstable")]
+    use zcash_client_backend::keys::sapling;
 
     #[cfg(feature = "unstable")]
-    use super::{
-        chain::{init::init_blockmeta_db, BlockMeta},
+    use crate::{
+        chain::init::init_blockmeta_db,
+        testing::{fake_compact_block, store_in_fsblockdb, AddressType},
         FsBlockDb,
     };
 
-    #[cfg(feature = "mainnet")]
```

Comment:
Never mind; it'll be easier to just rewrite anyway.

Reply:
When I initially performed the move, I thought I'd end up moving more of those methods _into_ `TestBuilder` / `TestRunner`. But it ended up being simpler to just call the existing methods (except for `insert_into_cache` etc. that were very small). I could probably remove the first commit and still have things work (it would be some effort, but doable).